app/services/course/material/zip_download_service.rb

```diff
@@ -1,12 +1,70 @@
 class Course::Material::ZipDownloadService
-  class << self
-    # Downloads the materials and zip them.
-    #
-    # @param [Course::Material::Folder] folder The folder containing the materials.
-    # @param [Array<Course::Material>] materials The materials to be downloaded.
-    # @return [String] The path to the zip file.
-    def download_and_zip(_folder, _materials)
-      # TODO: To implement
+  attr_accessor :materials, :folder, :base_dir
+
+  # @param [Course::Material::Folder] folder The folder containing the materials.
+  # @param [Array<Course::Material>] materials The materials to be downloaded.
+  def initialize(folder, materials)
```

Comment:
Think about the API from an external perspective. Why should any object need to know that the service needs to be initialised to use it? Just think about abstraction barriers, and you should understand where I'm coming from